src/mod_openssl.c

```diff
@@ -253,12 +316,17 @@ network_ssl_servername_callback (SSL *ssl, int *al, server *srv)
           ssl, SSL_dup_CA_list(hctx->conf.ssl_ca_file_cert_names));
         /* forcing verification here is really not that useful
          * -- a client could just connect without SNI */
-        mode = SSL_VERIFY_PEER;
         if (hctx->conf.ssl_verifyclient_enforce) {
-            mode |= SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
+            SSL_set_verify(ssl, SSL_VERIFY_PEER|SSL_VERIFY_FAIL_IF_NO_PEER_CERT,
+                           NULL);
+            SSL_set_verify_depth(ssl, hctx->conf.ssl_verifyclient_depth);
+        } else {
+            hctx->index = SSL_get_ex_new_index(0, "lighttpd-handler", NULL, NULL, NULL);
+            SSL_set_ex_data(ssl, hctx->index, hctx);
+
```

Comment:
Are the above two lines necessary?  (and is hctx->index member necessary?)
You use `hctx = (handler_ctx *) SSL_get_app_data(ssl);` in `verify_callback()`, so the above two lines appear to be unused.